references/detection/train.py

```diff
@@ -42,10 +42,16 @@ def copypaste_collate_fn(batch):
 
 def get_dataset(is_train, args):
     image_set = "train" if is_train else "val"
-    paths = {"coco": (args.data_path, get_coco, 91), "coco_kp": (args.data_path, get_coco_kp, 2)}
-    p, ds_fn, num_classes = paths[args.dataset]
-
-    ds = ds_fn(p, image_set=image_set, transforms=get_transform(is_train, args), use_v2=args.use_v2)
+    num_classes, mode = {"coco": (91, "instances"), "coco_kp": (2, "person_keypoints")}[args.dataset]
+    with_masks = "mask" in args.model
```

Comment:
Maybe we should remove the `get_masks` parameter of `get_coco()` and just make it part of the `mode` parameter, i.e. allow "instances", "instances_masks", and "keypoints".

ooooor just completely change the parametrization of `get_coco()` to something more friendly.

In addition to that, the way we magically specify keypoints or masks (through dataset name or through model name) is pretty terrible right now.

I feel like we should address that for good, but I'd prefer doing it in another PR

Reply:
Yes, I agree that it would be better that `args.dataset` could parametrize data type instead of the model name.
`args.dataset` can be "coco, "coco_kp" and "coco_masks", for example.